@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,194 @@ export namespace MapViewUtils {
+        camera.position.sub(rotationTargetWorld);
+        camera.position.applyQuaternion(headingQuat);
+        camera.position.add(rotationTargetWorld);
+    }
+
+    /**
+     * @hidden
+     * @internal
+     *
+     * Clamps the supplied `deltaTilt` to the `maxTiltAngle` supplied. Note, when a non-zero offset
+     * is applied, we apply another max angle of 89 degrees to the rotation center to prevent some
+     * corner cases where the angle at the rotation center is 90 degrees and therefore intersects
+     * the geometry with the near plane.
+     */
+    function computeClampedDeltaTilt(
+        mapView: MapView,
+        offsetY: number,
+        deltaTilt: number,
+        maxTiltAngle: number,
+        mapTargetWorld: THREE.Vector3,
+        rotationTargetWorld: THREE.Vector3,
+        tiltAxis: THREE.Vector3
+    ): number {
+        const camera = mapView.camera;
+        const projection = mapView.projection;
+
+        const tilt = extractTiltAngleFromLocation(projection, camera, mapTargetWorld, tiltAxis);
+        if (tilt + deltaTilt < 0) {
+            // Clamp the final tilt to 0
+            return -tilt;
+        } else if (deltaTilt <= 0) {
+            // Reducing the tilt isn't clamped (apart from above).
+            return deltaTilt;
+        } else if (mapTargetWorld.equals(rotationTargetWorld) || offsetY < 0) {
+            // When the rotation target is the center, or the offsetY is < 0, i.e. the angle at the
+            // `mapTargetWorld` is always bigger, then we have a simple formula
+            return MathUtils.clamp(deltaTilt + tilt, 0, maxTiltAngle) - tilt;
+        }
+
+        const rotationCenterTilt = extractTiltAngleFromLocation(
+            projection,
+            camera,
+            rotationTargetWorld!,
+            tiltAxis
+        );
+
+        const maxRotationTiltAngle = THREE.MathUtils.degToRad(89);
+
+        // The rotationCenterTilt may exceed 89 degrees when for example the user has tilted to 89
+        // at the mapTargetWorld, then choose a rotation center target above the mapTargetWorld,
+        // i.e. offsetY > 0. In such case, we just return 0, i.e. we don't let the user increase
+        // the tilt (but it can decrease, see check above for "deltaTilt <= 0").
+        if (rotationCenterTilt > maxRotationTiltAngle) {
+            return 0;
         }
 
-        const posBackup = cache.vector3[1].copy(camera.position);
-        const quatBackup = cache.quaternions[1].copy(camera.quaternion);
+        // This is used to find the max tilt angle, because the difference in normals is needed
+        // to correct the triangle used to find the max tilt angle at the rotation center.
+        let angleBetweenNormals = 0;
+        if (projection === sphereProjection) {
+            const projectedRotationTargetNormal = projection
+                .surfaceNormal(rotationTargetWorld, cache.vector3[0])
+                .projectOnPlane(tiltAxis);
```

Comment:
Here you're taking the surface normal at the rotation target and projecting it to the tilt plane. Instead, you should project the rotation target on the tilt plane, and the get the surface normal at that location.

Reply:
Why? The surface normal on the plane and the projected surface normal to the plane have the same direction.

Reply:
But... forgot to normalize()